e2e/suite1/specs/variables/new-query-variable.ts

```diff
@@ -105,7 +105,11 @@ describe('Variables - Add variable', () => {
 
     e2e.pages.Dashboard.SubMenu.submenuItemLabels('a label').should('be.visible');
 
-    e2e.pages.Dashboard.SubMenu.submenuItemValueDropDownValueLinkTexts('A').eq(1).should('be.visible').click();
+    e2e.pages.Dashboard.SubMenu.submenuItemValueDropDownValueLinkTexts('A')
+      .should('be.visible')
```

Comment:
@hugohaggmark this test seemed to be a bit flaky - it passed before I fixed the other tests with the v6 breaking changes but then failed after, even though I didn't change anything (with this). When running locally (both headless and not), I could never get this to fail like it did in the build but thought I'd add in this additional assertion, just in case. Do you reckon it's fine to do so?

Reply:
Yes, this one is unfortunately a bit flaky. Can you try and add 500ms to 
https://github.com/grafana/grafana/blob/027944dfd6db6142e9415e4e5c7b135a7a368be2/e2e/suite1/specs/variables/new-query-variable.ts#L102
and
https://github.com/grafana/grafana/blob/027944dfd6db6142e9415e4e5c7b135a7a368be2/e2e/suite1/specs/variables/new-query-variable.ts#L170

Reply:
If we use specific `wait`s before `should` assertions, are we pretty much prolonging the `should` assertion timeout? Are the `should`s not enough to avoid flakiness?

Reply:
I wasn't aware you could change the individual timeouts, if we can change the timeout on those that would be a lot better. Thank you.

Reply:
I'm reading up on Cypress (I've not used it before, so bear with me 🙏 ) to have a better understanding of our timeouts and correct me if I'm wrong, but it looks like we have the following options:

- Increase the timeouts for single assertions. In this case, increase the timeout for the `should` assertions after the `wait`s suggested: https://docs.cypress.io/api/commands/should.html#Timeouts

- Increase the timeouts for all assertions for all tests by changing the `defaultCommandTimeout` in the config: https://docs.cypress.io/guides/references/configuration.html#Timeouts

- Increase the timeouts for all assertions for just one test: https://stackoverflow.com/questions/55019061/increase-timeout-for-specific-its-method-of-cypress-io

Which do you think we should implement? Is there a reason why we wouldn't want to increase the default timeout to prevent flakiness across all tests?

Reply:
Yes, we need to keep everything as fast as possible, just adding those waits kind of made me very sad. We should make exceptions at micro level if you ask me, so I'd increase timeout for the individual assertions that fail here and remove the waits.

Reply:
Awesome, thanks! I will increase the timeout in the individual assertions to 4500 (for the default 4000 + the 500 suggested)

Reply:
With the previous waits it should be 5000, correct? Or are you keeping the waits. Sorry for the confusion.

Reply:
I was thinking instead of the `wait`s we can pass in the individual timeouts and if the default command timeout for cypress is 4000 and we wanted to have a wait of 500, we would be able to add in a timeout of 4500 for both `should` assertions that come after?

```
// instead of this e2e().wait(500);
// the should assertion has default of 4000 but we want to add 500 to make it 4500 below
e2e.components.BackButton.backArrow().should('be.visible').click({ force: true });
```

Let me know if I've misunderstood?

I also had a look at passing in the individual timeouts and it looks like we would need to update the wrappers here: https://github.com/grafana/grafana/blob/master/packages/grafana-e2e/src/support/types.ts#L60 with

```
return e2e().get(Selector.fromAriaLabel(value), { timeout });
return e2e().get(selector, { timeout })
return e2e().get(Selector.fromAriaLabel(selector, { timeout }));
```

Just to double check, is my understanding correct? If that's the case then I can add the `wait`s as suggested here first to help with the flakiness and then I can create a separate PR for allowing individual timeouts to be passed into our wrappers? Just to keep the PR smaller and I don't want to block other work that's waiting on this PR 😄

Reply:
Oh, I misread your previous comment and thought you meant to add the `wait`s but didn't realise they were already there 🤦‍♀️  Yup, you are right, it would be 5000 😄

Reply:
Increased the wait and it still seems flaky 😔

Reply:
Regarding the order of PRs, I totally agree with you, that sounds like a great approach. Regarding the flakiness with the increase, try another 500ms if that doesn't work then there is probably something else wrong and I can help look into it too.

Reply:
Awesome, thanks!

I've increased it and it looks like it still fails at the exact same place:

```
1143 | 1) Variables - Add variable
1144 | adding a multi value query variable:
1145 | AssertionError: Timed out retrying after 4000ms: Expected to find element: `1`, but never found it. Queried from element: <a.css-10l6kcd>
1146 | at Context.eval (http://end-to-end-tests-server:3001/__cypress/tests?p=../../e2e/suite1/specs/variables/new-query-variable.ts:1855:87)
```

I can't reproduce it locally but I haven't merged latest master in yet, so I'll do that and see if I can get it failing

Reply:
Merged with latest master locally to test and I still can't reproduce locally:

![Screenshot 2021-01-29 at 09 55 14](https://user-images.githubusercontent.com/36230812/106260759-a8766280-6218-11eb-8d62-2d6471aa396f.png)

The error in the build never shows up and it's fine with the `eq(1)` selector:

![Screenshot 2021-01-29 at 09 57 24](https://user-images.githubusercontent.com/36230812/106260809-b9bf6f00-6218-11eb-9339-0aa942a664a5.png)

Any suggestions on how I can debug this further? In the e2e tests I've been debugging for our enterprise plugins, I was always able to see the same errors locally as the build. Is there anything the build does differently here, in particular? 🤔

Reply:
Sorry for the late response, I've been in meetings. No, this flakiness is hard to repro locally. I'm taking it for a spin now.

Reply:
Thank you! I tried removing the waits locally to see if I can make it fail but they still passed 😓 

Is this test failing more for this PR in particular or does it tend to fail a lot in other PRs too? I opened another PR this morning and the build for that was fine - I was wondering if maybe v6 of cypress makes it more flaky for some reason?